sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/InMemoryCatalog.scala

```diff
@@ -253,6 +266,7 @@ class InMemoryCatalog(
           throw new SparkException(s"Unable to rename table $oldName to $newName as failed " +
             s"to rename its directory $oldDir", e)
       }
+      oldDesc.table = oldDesc.table.withNewStorage(locationUri = Some(newDir.toUri.toString))
```

Comment:
We can hide the all hive specific details in HiveExternalCatalog? I see there is many code place where we try to add a locationUri for managed table. It doesn't seems necessary if the locationUri can be deduced from table name. 

We can always do some conversion in HiveExternalCatalog to change CatalogTable to hive understandable catalog.
 
```
private def saveTableIntoHive(tableDefinition: CatalogTable, ignoreIfExists: Boolean): Unit = {
```